Chatkit/Auth/TokenProvider.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  TokenProvider.swift
+//  Chatkit
+//
+//  Created by Mike Pye on 06/12/2019.
+//  Copyright © 2019 Pusher Ltd. All rights reserved.
+//
+// STUB FILE. This file demonstrates the desired interface, but contains to implementation
+// TODO: Move many of these types to PusherPlatform
+
+import Foundation
+import PusherPlatform
+
+// TODO: Temporary
+extension String: Error { }
+
+// Re-export so that PusherPlatform import is not required in client code
+public typealias TokenProvider = PPTokenProvider
+public typealias TokenProviderResult = PPTokenProviderResult
+
+/// A factory for `TokenProvider`s.
+///
+/// The methods on this object create token provider instances which can interface with an HTTP endpoint in your backend, or the Chatkit Test Token Provider (for use during development)
+public struct ChatkitTokenProviders {
```

Comment:
Why is this prefixed with `Chatkit`? This should be derived from the module name.

Reply:
No good reason. Should we change the module name to match (as Marek suggests), or drop the prefix from the things in here?

Reply:
Is the `struct` even required here if there are only `static func`? I have a very limited knowledge in Swift and may be asking silly questions, coming from the Kotlin world ;)

Reply:
Currently the module is defined as `PusherChatkit`. What is the suggested new name?

When it comes to the token provider, at least in Swift, it would feel more natural to call the struct `TokenProvider`.

Reply:
@marekoid It can be a different data structure as well, but you need to have something. `struct` is very lightweight and is probably a good choice from Swift perspective.

Reply:
Oh then it sounds like there are no top-level functions like in Kotlin ;)

Reply:
You can use top-level/global functions, but that is rarely used. Very often `static` or `class` variables\methods are wrapped in structs\classes to provide some more context.

Reply:
Here function name provides full context.

Reply:
But there is no importing of anything smaller than an entire module, and no namespacing, so wrapping them in a struct is a way to offer some namespacing.

Reply:
Still there won't be a clash on `createHttpTokenProvider` function name within our module.